src/init_engine.rs

```diff
@@ -0,0 +1,136 @@
+use core::marker::PhantomData;
+
+use embedded_hal::delay::DelayNs;
+
+use crate::{
+    dcs::DcsCommand,
+    interface::{Interface, InterfaceAsync, InterfaceKind},
+    models::ModelInitError,
+};
+
+/// todo
+pub trait InitEngine {
+    // required for interface checks in Model::init
+    const INTERFACE_KIND: InterfaceKind;
+    type Error: core::fmt::Debug;
+    /// Queue a raw command with optional parameters
+    fn queue_command_raw(&mut self, command: u8, args: &[u8]) -> Result<(), Self::Error>;
+
+    fn queue_command(&mut self, command: impl DcsCommand) -> Result<(), Self::Error>;
+
+    fn queue_delay_us(&mut self, us: u32) -> Result<(), Self::Error>;
+
+    fn pop_command(&mut self) -> Option<QueuedInitCommand>;
+}
+
+pub struct InitEngineSync<'delay, DI, DELAY> {
+    di: DI,
+    delay: &'delay mut DELAY,
+}
+
+pub struct InitEngineAsync<DI> {
+    queue: heapless::Deque<QueuedInitCommand, 24>,
+    _di: PhantomData<DI>,
+}
+
+pub enum QueuedInitCommand {
+    RawDcs(u8, [u8; 16]),
```

Comment:
This needs an additional parameter for the data length, otherwise all DCS commands will send 16 bytes of parameters. I'm not sure if all controllers handle excess bytes gracefully, but I think it's better not to send them.